src/c-list.h

```diff
@@ -323,41 +331,48 @@ static inline CList *c_list_last(CList *list) {
              (_iter) != (_list);                                                \
              _iter = (_iter)->next)
 
+#if _C_LIST_HAS_TYPEOF
```

Comment:
I would actually keep them defined. It keeps this simple and the compilers should all be fine. Any particular reason to only define them if they are supported?

If we avoid these guards, we can drop the `_C_LIST_HAS_TYPEOF` from the public header, and avoid hard-coding compiler-specifics.